bridge_common/validateapi.py

```diff
@@ -0,0 +1,191 @@
+            schemaFile, str(e)))
+
+    return (True, config)
+
+
+def validateApiWithYamlConf(apiJob, yamlConf):
+    if 'valid_objects' in yamlConf:
+        if apiJob['object_type'] not in yamlConf['valid_objects']:
+            return (False, "Invalid object type or object type not supported!")
+
+    # Check flow exists
+    atom = yamlConf['object_details'][apiJob['object_type']]['atoms']
+    if apiJob['flow'] not in atom:
+        return (False, "Invalid flow type or flow not supported!")
+
+    # check whether given api-job has any arguments to check
+    # or any arguments defined for this purticular job in the yaml.
+    flow = atom[apiJob['flow']]
+    if 'local_inputs' not in flow or 'attributes' not in apiJob:
+        return (True, "")
+
+    # fetch the list of optional and required defined list from yaml
+    defReqLst = [d for d in flow['local_inputs'] if not isinstance(d, list)]
+    defOptLst = [d[0] for d in flow['local_inputs'] if isinstance(d, list)]
+
+    # optain the given optional and remaining input param
+    # from the given api job struct
+    gvnLst = [i for i in
+              apiJob['attributes'].keys() if i not in defOptLst]
+
+    # check whether any required argument is missing.
+    # check the list of arguments given by api-job with the
+    # list of required argument mentioned under flow tag in the yaml config.
+    missingInputParm = set(defReqLst).difference(
+        set(gvnLst))
+    if missingInputParm != set():
+        return (False,
+                "Missing input argument(s) %s" % (list(missingInputParm)))
+
+    # check whether all the given arguments are defined in the yaml file.
+    missingConfigParm = set(gvnLst).difference(
+        set(defReqLst))
+    if missingConfigParm != set():
+        return (False, "Input argument(s) not defined in yaml file: %s" % (
+            list(missingConfigParm)))
+
+    # verify whether argument value align with required type.
+    # collect the attribute name and its type and check with the
+    # type mentioned in the yaml file for the variable name.
+    attrs = yamlConf['object_details'][apiJob['object_type']]['attrs']
+    for inputParm, inputVal in apiJob['attributes'].items():
+        # Some arguments does not required any purticular type.
+        # Contine validating next parm if the type not defined in yaml.
+        if inputParm not in attrs:
+            continue
+        if not attrs[inputParm]['type'] in PRIMITIVE_TYPES.keys():
+            return (False, "Invalid parameter type :%s" % (
+                attrs[inputParm]['type']))
+        # Check the given value type and the required type is matching
+        if not PRIMITIVE_TYPES[attrs[inputParm]['type']](inputVal):
+            return (False, "Invalid parameter type: "
+                    + "%s. Expected value type is: %s" % (
+                        inputParm, attrs[inputParm]['type']))
+    return(True, "")
+
+
+class SdsOperations(object):
```

Comment:
I feel, this class has to provide option to validate a job for specific schema instead of loading everything. Because this module/utility will be used in different components like: ceph-bridge, gluster-bridge & node-agent. It would be better to verify across a single schema file instead of loading everything.